src/Microsoft.OData.Core/UriParser/Aggregation/AggregationMethod.cs

```diff
@@ -9,7 +9,31 @@ namespace Microsoft.OData.UriParser.Aggregation
     /// <summary>
     /// Enumeration of methods used in the aggregation clause
     /// </summary>
-    public enum AggregationMethod
+    public sealed class AggregationMethod

```

Comment:
Looking at the WebAPI side of this change, we might need to add the definition alongside the kind in AggregateExpression to avoid a breaking change; if we change the type of AggregateExpression.Method out from under people, it would be a breaking change.  Maybe add AggregateExpression.MethodDefinition and let AggregateExpression.Method return "custom" for the new custom ones.